### DISCUSSION TITLE
Trying to collaborate

### DISCUSSION THREAD

**Discussion by andra961** [react-native-modal/react-native-modal#668] [Q&A]
Hi everyone!This would be my first contribution to a community package and i'm having trouble to understand the workflow which would lead to a pull request. I forked the repository, then cloned it in a folder inside my project where i'd like to test the changes i will make. Once cloned i do a `npm install` inside the forked repo and then `npm run build`. The problem is that during the build i get lots of `error TS1005: ';' expected` and `error TS1005: '=' expected`. My plan if the build had success would have been to make changes in the .ts files, build again through build command and so on until i commit. Have i understand correctly the process?if not could you point out where i misunderstood?thx in advance

**Comment by mmazzarolo** [comment]
Hi Andra! :)
Unfortunately I can't pull down the repo and give it a test, but the flow I recommend using for working on this repo is available in the [CONTRIBUTING.md](https://github.com/react-native-modal/react-native-modal/blob/master/.github/CONTRIBUTING.md) documentation 👍 Let me know how it goes.

**Reply by andra961** [reply]
Yeah unfortunately i still get those errors when i try to build with typescript. I'm sure i'm using the local version in the node modules which i installed with npm install...

**Reply by andra961** [reply]
I get this error when i build: 
```
src/modal.tsx:538:35 - error TS2339: Property 'includes' does not exist on type 'Direction[]'.

538       ? this.props.swipeDirection.includes(direction)
```

When i run the example android app i get `no stack` error

**Reply by mmazzarolo** [reply]
Are you sure you're using the right version of TS? (meaning the one included in the project: https://github.com/react-native-modal/react-native-modal/blob/d0bf96bfe84f2227820b9d8086a79e0781c0c385/package.json#L78)

**Comment by andra961** [comment]
Yes i even tried to run it directly from node_modules

**Reply by andra961** [reply]
Idk so far nothing seems to be working in the repository  i cloned. Is someone actually using the example app in the repo or is it not mantained anymore and normal that it gives me "no stack" error when i try to launch it (without touching the cloned repo default settings)?

**Reply by ancyrweb** [reply]
Hey Andra, i'm going to investigate. The repository definetely needs some update, it hasn't correctly followed the last changes in the ecosystem. I'll work on a branch.

**Reply by andra961** [reply]
> Hey Andra, i'm going to investigate. The repository definetely needs some update, it hasn't correctly followed the last changes in the ecosystem. I'll work on a branch.

Thank you so much!

**Reply by ancyrweb** [reply]
Can you check the `upgrade` branch and install dependencies with `yarn install` on both the main folder and the `Example` folder ? It should work now.

**Reply by andra961** [reply]
> Can you check the `upgrade` branch and install dependencies with `yarn install` on both the main folder and the `Example` folder ? It should work now.

i installed them in both folder with npm(should be of couse the same), then did `npm run dev` in main folder, then in `example` folder `npx react-native start` (tried also `npx react-native start --reset-cache`) followed by `npx react-native run-android`(the usual RN flow). No typescript in modal folder compiles without errors(great!). However the app still doesn't launch. I get in metro terminal this error:

C:\Users\user\Desktop\react-native-modal\example\node_modules\metro-hermes-compiler\src\emhermesc.js:77
          throw ex;
          ^

Error: ENOENT: no such file or directory, watch 'C:\Users\user\Desktop\react-native-modal\example\node_modules\react-native-gesture-handler\android\build\kotlin\compileDebugKotlin\caches-jvm'
    at FSWatcher.<computed> (internal/fs/watchers.js:218:26)
    at Object.watch (fs.js:1525:34)
    at NodeWatcher.watchdir (C:\Users\user\Desktop\react-native-modal\example\node_modules\sane\src\node_watcher.js:159:22)
    at NodeWatcher.<anonymous> (C:\Users\user\Desktop\react-native-modal\example\node_modules\sane\src\node_watcher.js:295:18)
    at FSReqCallback.oncomplete (fs.js:184:5) {
  errno: -4058,
  syscall: 'watch',
  code: 'ENOENT',
  path: 'C:\\Users\\user\\Desktop\\react-native-modal\\example\\node_modules\\react-native-gesture-handler\\android\\build\\kotlin\\compileDebugKotlin\\caches-jvm',
  filename: 'C:\\Users\\user\\Desktop\\react-native-modal\\example\\node_modules\\react-native-gesture-handler\\android\\build\\kotlin\\compileDebugKotlin\\caches-jvm'
}

**Reply by ancyrweb** [reply]
Did you install dependencies using yarn ?

**Reply by andra961** [reply]
> Did you install dependencies using yarn ?

No i used npm but it should be the same. Later i can try with yarn too. But can you please answer a question(i'm trying to learn)?so the contributing flow would be testing changes i make to the package in the example app, i get it. But what if i'd like to instead testing them on my existing app before committing?would i have to copy paste the cloned repo in my app root folder and just import it like this: "import Modal from "../react-native-modal/index/index.js"?

**Reply by ancyrweb** [reply]
You might want to check `yarn link` or equivalent `npm link` method, whose purpose is to enable development of packages. Since we are using yarn with a lockfile that holds the exact version numbers, you should use it to work on the project.

**Reply by andra961** [reply]
> You might want to check `yarn link` or equivalent `npm link` method, whose purpose is to enable development of packages. Since we are using yarn with a lockfile that holds the exact version numbers, you should use it to work on the project.

Tried with yarn too but i get the same error above unfortunately :(

**Reply by ancyrweb** [reply]
It looks like your installation is incorrect, try to delete the node_modules entirely and install from yarn. Eventually clear yarn cache ?

**Reply by andra961** [reply]
> It looks like your installation is incorrect, try to delete the node_modules entirely and install from yarn. Eventually clear yarn cache ?

Tried it many times with both npm and yarn, also resetting cache, same error. Which version of node are you using, also which OS?i have node 14.15.4 on Windows
